### ISSUE TITLE
cannot inherit from a parameters 

### ISSUE THREAD

**Issue by samsja** [python/mypy#14185]
**Bug Report**

It seems that mypy does not allow to inherit from a parameter when creating a function on the fly.

**To Reproduce**

```python
from typing import Type


def f(cls: Type) -> Type:


    class A(cls):
        ...
    
    return A

>>>file.py:6: error: Variable "cls" is not valid as a type  [valid-type]
>>>file.py:6: note: See https://mypy.readthedocs.io/en/stable/common_issues.html#variables-vs-type-aliases
>>>file.py:6: error: Invalid base class "cls"  [misc]

```

**Expected Behavior**

Mypy should not complain

**Actual Behavior**

Mypy complain and say

```
>>>file.py:6: error: Variable "cls" is not valid as a type  [valid-type]
>>>file.py:6: note: See https://mypy.readthedocs.io/en/stable/common_issues.html#variables-vs-type-aliases
>>>file.py:6: error: Invalid base class "cls"  [misc]
```

**Your Environment**

- Mypy version used:`0.990
- Python version used: 3.9.10

**Comment by thomkeh** [comment]
I believe that's fixed by #14135 which has been merged to master but hasn't been released yet.

**Comment by AlexWaygood** [comment]
> I believe that's fixed by #14135 which has been merged to master but hasn't been released yet.
> 
> (I haven't tried it out though.)

Correct, this is already fixed on `master`. I can reproduce with https://github.com/python/mypy/commit/c660354846688ff8158d0f0178eb298171b74f5b, but not with https://github.com/python/mypy/commit/3c5f368872413fae867bf4a2dff7781cbf547459. The fix will be released as part of mypy 1.0, coming soon!

**Comment by AlexWaygood** [comment]
> Just tried with this version it work indeed. Thanks for the amazing work !

You have @ilevkivskyi to thank :)
